pkg/config/db.go

```diff
@@ -0,0 +1,158 @@
+		MaxLocksPerTransaction        *uint   `toml:"max_locks_per_transaction"`
+		MaxParallelMaintenanceWorkers *uint   `toml:"max_parallel_maintenance_workers"`
+		MaxParallelWorkers            *uint   `toml:"max_parallel_workers"`
+		MaxParallelWorkersPerGather   *uint   `toml:"max_parallel_workers_per_gather"`
+		MaxReplicationSlots           *uint   `toml:"max_replication_slots"`
+		MaxSlotWalKeepSize            *string `toml:"max_slot_wal_keep_size"`
+		MaxStandbyArchiveDelay        *string `toml:"max_standby_archive_delay"`
+		MaxStandbyStreamingDelay      *string `toml:"max_standby_streaming_delay"`
+		MaxWalSize                    *string `toml:"max_wal_size"`
+		MaxWalSenders                 *uint   `toml:"max_wal_senders"`
+		MaxWorkerProcesses            *uint   `toml:"max_worker_processes"`
+		SessionReplicationRole        *string `toml:"session_replication_role"`
+		SharedBuffers                 *string `toml:"shared_buffers"`
+		StatementTimeout              *string `toml:"statement_timeout"`
+		WalKeepSize                   *string `toml:"wal_keep_size"`
+		WalSenderTimeout              *string `toml:"wal_sender_timeout"`
+		WorkMem                       *string `toml:"work_mem"`
+	}
+
+	db struct {
+		remoteDb
+		Image        string `toml:"-"`
+		Port         uint16 `toml:"port"`
+		ShadowPort   uint16 `toml:"shadow_port"`
+		MajorVersion uint   `toml:"major_version"`
+		Password     string `toml:"-"`
+		RootKey      string `toml:"-" mapstructure:"root_key"`
+		Pooler       pooler `toml:"pooler"`
+		Seed         seed   `toml:"seed"`
+	}
+
+	seed struct {
+		Enabled      bool     `toml:"enabled"`
+		GlobPatterns []string `toml:"sql_paths"`
+		SqlPaths     []string `toml:"-"`
+	}
+
+	pooler struct {
+		Enabled          bool     `toml:"enabled"`
+		Image            string   `toml:"-"`
+		Port             uint16   `toml:"port"`
+		PoolMode         PoolMode `toml:"pool_mode"`
+		DefaultPoolSize  uint     `toml:"default_pool_size"`
+		MaxClientConn    uint     `toml:"max_client_conn"`
+		ConnectionString string   `toml:"-"`
+		TenantId         string   `toml:"-"`
+		EncryptionKey    string   `toml:"-"`
+		SecretKeyBase    string   `toml:"-"`
+	}
+)
+
+// Compare two pointers values handling the nil case
+func isPointerValueEquals[T comparable](a, b *T) bool {
+	if a == nil && b == nil {
+		return true
+	}
+	if a == nil || b == nil {
+		return false
+	}
+	return *a == *b
+}
+
+// Compare two db config, if changes requires restart return true, return false otherwise
+func requireDbRestart(a *db, b *db) bool {
+	return !isPointerValueEquals(a.MaxConnections, b.MaxConnections) ||
+		!isPointerValueEquals(a.MaxWorkerProcesses, b.MaxWorkerProcesses) ||
+		!isPointerValueEquals(a.MaxParallelWorkers, b.MaxParallelWorkers) ||
+		!isPointerValueEquals(a.MaxWalSenders, b.MaxWalSenders) ||
+		!isPointerValueEquals(a.MaxReplicationSlots, b.MaxReplicationSlots) ||
+		!isPointerValueEquals(a.SharedBuffers, b.SharedBuffers)
+}
```

Comment:
**remark**

I see that the update endpoint allow to pass a `restart` parameter so it's externally controlled. I guess we want to gives users full control about wether to restart or not when they change each field one by one. But when we update all the parameters at once (like branching) we wanna automatically restart if needed.